presto-spi/src/main/java/com/facebook/presto/spi/type/P4HyperLogLogType.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.type;
+
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import io.airlift.slice.Slice;
+
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+
+public class P4HyperLogLogType
+    extends AbstractVariableWidthType
+{
+    public static final P4HyperLogLogType P4HYPER_LOG_LOG = new P4HyperLogLogType();
+    private HyperLogLogType proxy;
+
+    @JsonCreator
+    public P4HyperLogLogType()
+    {
+        super(parseTypeSignature(StandardTypes.P4HYPER_LOG_LOG), Slice.class);
+        proxy = new HyperLogLogType();
+    }
+
+    @Override
+    public void appendTo(Block block, int position, BlockBuilder blockBuilder)
+    {
+        proxy.appendTo(block, position, blockBuilder);
```

Comment:
you can directly reference the static instance of StandardTypes.HYPER_LOG_LOG instead of storing the proxy.